Src/LiquidProjections/Statistics/ProjectorStats.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+
+namespace LiquidProjections.Statistics
+{
+    /// <summary>
+    /// Contains statistics and information about a particular projector.
+    /// </summary>
+    public class ProjectorStats
+    {
+        private readonly ConcurrentDictionary<string, Property> properties =
+            new ConcurrentDictionary<string, Property>();
+
+        private readonly BlockingCollection<Event> events = new BlockingCollection<Event>();

```

Comment:
`BlockingCollection` is designed for producer-consumer scenarios. It's not any more optimal than a `List` with `lock` around its usages (in fact it's less optimal).